lambdas/shared/t4_lambda_shared/utils.py

```diff
@@ -61,8 +62,18 @@ def get_quilt_logger():
 
 
 def get_available_memory():
-    """how much virtual memory is available to us (bytes)?"""
-    from psutil import virtual_memory
+    """how much virtual memory is available to us (in bytes)"""
+    try:
+        # this import fails in AWS Lambda in spite of requirements.txt
```

Comment:
Huh? Why does it fail? And if that's the case, how does any lambda that calls `get_available_memory` work at all?

Though if it does fail, I'd just drop `psutil` completely in favor of the env var.

Reply:
Yeah this was not the problem. Though I don't see need to revert this new code but I can if you want. The problem is that the first row-group is huge but the new pyarrow has an iterator instead of reading the whole row group, so I'm gonna try that.